src/core/blockchain/near/transaction.ts

```diff
@@ -95,13 +131,94 @@ export class NearTransactionUtils extends AbstractBlockchainTransactionUtils {
         };
     }
 
+    public async buildPosTransaction(
+        tx: IPosTransaction,
+        transactionType: PosBasicActionType
+    ): Promise<IBlockchainTransaction[]> {
+        const client = Near.getClient(tx.chainId);
+
+        const transactions: IBlockchainTransaction[] = [];
+
+        switch (transactionType) {
+            case PosBasicActionType.DELEGATE: {
+                for (const validator of tx.validators) {
+                    const txDelegate: IPosTransaction = cloneDeep(tx);
+
+                    const res = await new ApiClient().validators.getBalance(
+                        tx.account,
+                        client.chainId.toString()
+                    );
+
+                    let hasToDeposit = false;
+
+                    if (
+                        new BigNumber(tx.amount).isGreaterThan(new BigNumber(res.balance.unstaked))
+                    ) {
+                        hasToDeposit = true;
+
+                        const depositAmount = new BigNumber(tx.amount).minus(
+                            new BigNumber(res.balance.unstaked)
+                        );
+
+                        // Deposit
+                        const depositTx: IBlockchainTransaction = await (client as NearClient).staking.deposit(
+                            txDelegate,
+                            validator
+                        );
+
+                        depositTx.amount = depositAmount.toFixed();
+                        transactions.push(depositTx);
+                    } else {
+                        // no need to deposit
+                    }
+
+                    // Stake
+                    const stakeTx: IBlockchainTransaction = await (client as NearClient).staking.stake(
+                        txDelegate,
+                        validator
+                    );
+                    if (hasToDeposit) {
```

Comment:
no need for this if... if transactions.length is bigger then 0 it will increase, otherwise its 0